lib/src/utils/copy_map.dart

```diff
@@ -1,22 +1,20 @@
 /// The database always gives back an `_InternalLinkedHasMap<dynamic, dynamic>`.
 /// This creates a deep copy of the json and makes sure that the format is
 /// always `Map<String, Object?>`.
-extension CopyMapExtension on Map {
-  Map<String, Object?> get copy {
-    final copy = Map<String, dynamic>.from(this);
-    for (final entry in copy.entries) {
-      copy[entry.key] = _castValue(entry.value);
-    }
-    return copy;
+Map<String, Object?> copyMap(Map map) {
+  final copy = Map<String, dynamic>.from(map);
+  for (final entry in copy.entries) {
+    copy[entry.key] = _castValue(entry.value);
   }
+  return copy;
+}
 
-  dynamic _castValue(dynamic value) {
-    if (value is Map) {
-      return value.copy;
-    }
-    if (value is List) {
-      return value.map(_castValue).toList();
-    }
-    return value;
+dynamic _castValue(dynamic value) {
```

Comment:
Some day I would like to return an Object? here and possibly not reassign to a Map of dynamics in the loop above, but for now there is no benefit, I guess.